app/services/workflow/save_installments_service.rb

```diff
@@ -65,6 +71,8 @@ def process
     attr_reader :params, :seller, :workflow, :preview_email_recipient
 
     def delete_removed_installments
+      return if params[:installments].nil?
```

Comment:
What is causing the need for checking `params[:installments]`'s presence? This is not necessarily a bad thing, but was there a bug we didn't catch until now, or do the changes in this branch introduce a flaw that we are catching here?